cachito/workers/pkg_managers/yarn.py

```diff
@@ -55,3 +64,293 @@ def get_yarn_proxy_repo_username(request_id):
     :rtype: str
     """
     return f"cachito-yarn-{request_id}"
+
+
+def _get_deps(yarn_lock, file_deps_allowlist):
+    """
+    Process the dependencies in a yarn.lock file and return relevant information.
+
+    Will convert non-registry dependencies to Nexus hosted dependencies. If there are any
+    non-registry dependencies, the yarn.lock file should later be modified to point to the
+    replacement Nexus dependencies. This function returns all the information necessary to
+    do that.
+
+    The list of dependencies returned by this function is in a format that can be used directly
+    as input to the general_js.download_dependencies function.
+
+    :param dict yarn_lock: parsed yarn.lock data
+    :param set file_deps_allowlist: an allow list of dependencies that are allowed to be "file"
+        dependencies and should be ignored since they are implementation details
+    :return: information about preprocessed dependencies and Nexus replacements
+    :rtype: (list[dict], dict)
+    :raise CachitoError: if the lock file contains a dependency from an unsupported location
+    """
+    deps = []
+    nexus_replacements = {}
+
+    for dep_identifier, dep_data in yarn_lock.items():
+        package = pyarn.lockfile.Package.from_dict(dep_identifier, dep_data)
+
+        if package.url:
+            source = package.url
+        elif package.relpath:
+            source = f"file:{package.relpath}"
+        else:
+            raise CachitoError(f"The dependency {package.name}@{package.version} has no source")
+
+        nexus_replacement = None
+
+        non_registry = not package.url or not _is_from_npm_registry(package.url)
+        if non_registry:
+            log.info("The dependency %r is not from the npm registry", f"{package.name}@{source}")
+            # If the file dependency is in the allowlist, do not convert it to a Nexus
+            # dependency, simply ignore it (it will also be ignored by the code that downloads
+            # dependencies later)
+            if package.relpath and normpath(package.relpath) in file_deps_allowlist:
+                log.info("The dependency %r is an allowed exception", package.name)
+            else:
+                nexus_replacement = _convert_to_nexus_hosted(package.name, source, dep_data)
+
+        dep = {
+            "bundled": False,  # yarn.lock does not seem to contain bundled deps at all
+            "dev": False,  # yarn.lock does not state whether a dependency is dev
+            # TODO: ^ this can be solved by cross-referencing with package.json
```

Comment:
It does not :/

We could either try to infer it from the package.json file or just ignore it. This looks good for now, but it would be nice to track this somehow so we can address it in the future.

Reply:
Made a story to track this internally